api/src/OpenTelemetry/Logging/Core.hs

```diff
@@ -120,19 +132,48 @@ emitLogRecord Logger {..} LogRecordArguments {..} = do
     SpanContext {traceId, spanId, traceFlags} <- getSpanContext currentSpan
     pure (traceId, spanId, traceFlags)
 
+  let logRecordAttributes =
+        LA.addAttributes
+          (loggerProviderAttributeLimits loggerProvider)
+          LA.emptyAttributes
+          attributes
+
+  when (LA.attributesDropped logRecordAttributes > 0) $ void logDroppedAttributes
+
   pure
     LogRecord
       { logRecordTimestamp = timestamp
       , logRecordObservedTimestamp
       , logRecordTracingDetails
-      , logRecordSeverityNumber = fmap mkSeverityNumber severityNumber
-      , logRecordSeverityText = severityText <|> (shortName . mkSeverityNumber =<< severityNumber)
+      , logRecordSeverityNumber = severityNumber
+      , logRecordSeverityText = severityText <|> (toShortName =<< severityNumber)
       , logRecordBody = body
       , logRecordResource = loggerProviderResource loggerProvider
       , logRecordInstrumentationScope = loggerInstrumentationScope
-      , logRecordAttributes =
-          addAttributes
-            (loggerProviderAttributeLimits loggerProvider)
-            emptyAttributes
-            attributes
+      , logRecordAttributes
+      }
+
+
+-- | WARNING: this function should only be used to emit logs from the hs-opentelemetry-api library. DO NOT USE this function in any other context.
+logDroppedAttributes :: (MonadIO m) => m (LogRecord Text)
+logDroppedAttributes = emitOTelLogRecord H.empty Warn "At least 1 attribute was discarded due to the attribute limits set in the logger provider."
+
+
+-- | WARNING: this function should only be used to emit logs from the hs-opentelemetry-api library. DO NOT USE this function in any other context.
+emitOTelLogRecord :: (MonadIO m) => H.HashMap Text LA.AnyValue -> SeverityNumber -> Text -> m (LogRecord Text)
```

Comment:
Helper function to remove boilerplate. Takes `Attributes`, a `SeverityNumber`, and a message because those are the only things that need to be set in this context at this point. `Attributes` may be unnecessary, but it forces the person writing the code that emits an OTel log to think about whether there are `Attributes` that could be attached.